### PR TITLE
Add top bars and improve the help popup

### PR DESCRIPTION
**Pull request by ThomasMPont**
## Motivation

The help button currently takes up a large portion of the top center of each main screen. By being so central, it attracts unnecessary attention from users.

The current help popup is difficult to read and uses the wrong colors.

Annex screens can't currently be added help popups.

The current top bar buttons from main screens annoyingly don't align with the help button and don't stick to the top of the screen.

### Screenshots

All screens have been changed in this PR. I'll only include a few here.

Here is the new look of the home screen.
image

Here is the quiz screen.
image

### Top bar

Both main and annex screen scaffolds now define a top bar which can be filled with `BasicButton()`s as parameters. If the screen defines a help popup, the top bar will also contain a help button.
In annex screens, the top bar automatically holds a back button.
The arrangement of these buttons is automatically handled by the `Screens.TopBar()` composable.

The top bar displays a separator line under it to distinguish it from scrollable screen content.

### Help popup

Help popup titles and text were adjusted to be clearer and to use the correct theme colors.

### Padding

The padding of some screens and of common screen elements was adjusted to account for the new top bar.
An issue with the padding of screens that use a `LazyColumn()` forced me to relegate top padding to each screen.

## Tests

Simple tests were written for each new element that needed to be defined in `Screens.kt`.

## Figma

The Figma design was updated to reflect all changes.

### Dictionary scrolling is hardcoded

For some reason, the dictionary uses an integer resource to determine which entry to scroll to when a letter is tapped. This rigid implementation breaks whenever the count of elements in the home screen changes. I've created a new task that fixes this.

### CODE REVIEW

File: app/src/main/res/values-fr/strings.xml [hunk 1]
```diff
@@ -176,12 +176,11 @@
     <string name="tip_t">Conseils : Faites un poing avec votre pouce coincé entre votre index et votre majeur.</string>
     <string name="tip_u">Conseils : Tenez votre index et majeur vers le haut et proches l’un de l’autre, tandis que les autres doigts sont repliés vers le bas.</string>
     <string name="tip_v">Conseils : Tenez votre index et majeur vers le haut et séparés en forme de V, tandis que les autres doigts sont repliés vers le bas.</string>
-    <string name="tip_w">Conseils : Tenez votre index, majeur et annulaire vers le haut, les écartant, tandis que les autres doigts sont repliés vers le bas.</string>
+    <string name="tip_w">Conseils : Tenez votre index, majeur et annuFlaire vers le haut, les écartant, tandis que les autres doigts sont repliés vers le bas.</string>
```
Comment (ziyadbagh) [inline app/src/main/res/values-fr/strings.xml:179]:
```suggestion
    <string name="tip_w">Conseils : Tenez votre index, majeur et annulaire vers le haut, les écartant, tandis que les autres doigts sont repliés vers le bas.</string>
```
I think the "F" was added by mistake.

File: app/src/main/java/com/github/se/signify/ui/screens/profile/ProfileScreen.kt
Comment (ziyadbagh) [inline app/src/main/java/com/github/se/signify/ui/screens/profile/ProfileScreen.kt:1]: LGTM

File: app/src/main/java/com/github/se/signify/ui/screens/home/HomeScreen.kt
Comment (ziyadbagh) [inline app/src/main/java/com/github/se/signify/ui/screens/home/HomeScreen.kt:1]: This is much more cleaner now.
Good job!!

File: app/src/androidTest/java/com/github/se/signify/ui/screens/profile/ProfileScreenTest.kt [hunk 2]
```diff
@@ -126,20 +126,7 @@ class ProfileScreenTest {
   fun dialogIsDisplayedWhenHelpBoxIsVisible() {
 
     // Assert that the help dialog is displayed when the help button is clicked
-    composeTestRule.onNodeWithTag("InfoButton").performClick()
-    composeTestRule.onNodeWithTag("InfoPopupContent").assertIsDisplayed()
-  }
-
-  @Test
-  fun dialogClosesWhenCloseButtonClicked() {
```
Comment (ziyadbagh) [inline app/src/androidTest/java/com/github/se/signify/ui/screens/profile/ProfileScreenTest.kt:134]: Is there a specific reason why this test was deleted? I’m sure it wouldn’t affect the overall coverage significantly, but I’m curious.

File: app/src/main/java/com/github/se/signify/ui/common/Screens.kt
Comment (ziyadbagh) [inline app/src/main/java/com/github/se/signify/ui/common/Screens.kt:1]: Everything seems well done here.

File: app/src/androidTest/java/com/github/se/signify/ui/common/ScreensTest.kt
Comment (ziyadbagh) [inline app/src/androidTest/java/com/github/se/signify/ui/common/ScreensTest.kt:1]: LGTM

**Review by ziyadbagh — DISMISSED** [review]
As discussed in our group, adding the spacer directly to the AnnexScreenScaffold seems more intuitive and could save us from adding it to each screen individually. I see you’ve created a task for that, but since it’s unlikely we’ll have new screens, we could just leave it as is.

File: app/src/main/res/values-fr/strings.xml (hunk 1 above)
Comment (ThomasMPont) [reply app/src/main/res/values-fr/strings.xml:179]: Yup, I've finally found out how to stop Android Studio from autosaving everything. I'll stop commiting random commands into files now.

File: app/src/androidTest/java/com/github/se/signify/ui/screens/profile/ProfileScreenTest.kt (hunk 2 above)
Comment (ThomasMPont) [reply app/src/androidTest/java/com/github/se/signify/ui/screens/profile/ProfileScreenTest.kt:134]: I've deleted the close button entirely. The test can't exist anymore.

**Comment by ThomasMPont** [comment]
> As discussed in our group, adding the spacer directly to the AnnexScreenScaffold seems more intuitive and could save us from adding it to each screen individually. I see you’ve created a task for that, but since it’s unlikely we’ll have new screens, we could just leave it as is.

I'll leave the task up as it also adds bottom padding to screens.

File: app/src/androidTest/java/com/github/se/signify/ui/screens/profile/ProfileScreenTest.kt (hunk 2 above)
Comment (ThomasMPont) [reply app/src/androidTest/java/com/github/se/signify/ui/screens/profile/ProfileScreenTest.kt:134]: Also there is no reason for the profile screen to be testing common elements anyway. I've let tests that do this be for now, when they didn't break. But ideally, only `Screens.kt` should be testing its elements, for modularity.

**Review by ziyadbagh — APPROVED** [review]

**Pull request merged**